src/core/bpf/socket_bind/socket-bind-skel.h

```diff
@@ -0,0 +1,14 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+/* The SPDX header above is actually correct in claiming this was
+ * LGPL-2.1-or-later, because it is. Since the kernel doesn't consider that
+ * compatible with GPL we will claim this to be GPL however, which should be
+ * fine given that LGPL-2.1-or-later downgrades to GPL if needed.
+ */
+
+/* libbpf is used via dlopen(), so rename symbols */
+#define bpf_object__open_skeleton sym_bpf_object__open_skeleton
+#define bpf_object__load_skeleton sym_bpf_object__load_skeleton
+#define bpf_object__destroy_skeleton sym_bpf_object__destroy_skeleton
```

Comment:
Think this approach should work?

Reply:
not sure I follow, this was not necessary with the previous hook, why is it necessary now?

Reply:
Previously this was being done manually with a regex in the python script, however I didn't see a clean way to do that with meson directly. This seems to work though.

Reply:
are these fixed? or can they change dynamically?

Reply:
Fixed I think, regex would likely break as well if they get changed.

Reply:
I meant, the regex catched any of these if new ones were added, while this is hard-coded - will there be new skeleton functions added? Is that possible?

Reply:
I think manual changes would typically also end up being required in a few places either way.